temporal-test-server/src/test/java/io/temporal/testserver/functional/NexusWorkflowTest.java

```diff
@@ -54,432 +54,1039 @@
 public class NexusWorkflowTest {
   @Rule
   public SDKTestWorkflowRule testWorkflowRule =
-      SDKTestWorkflowRule.newBuilder().setDoNotStart(true).build();
+      SDKTestWorkflowRule.newBuilder()
+          .setDoNotStart(true)
+          .setWorkflowTypes(EchoNexusHandlerWorkflowImpl.class)
+          .build();
+
+  private final Payload defaultInput =
+      Payload.newBuilder().setData(ByteString.copyFromUtf8("input")).build();
+  private Endpoint testEndpoint;
+  private final String testService = "test-service";
+  private final String testOperation = "test-operation";
 
   @Before
-  public void checkExternal() {
+  public void setup() {
     // TODO: remove this skip once 1.25.0 is officially released and
     // https://github.com/temporalio/sdk-java/issues/2165 is resolved
     assumeFalse(
```

Comment:
If you remove this and set the test rules to use an external server. Assuming your server version supports nexus do these tests pass?

Reply:
I don't think the async ones will pass because of how the callback is currently handled. For the test server, I'm just encoding the operation reference; for an external server we'd need to generate the correct HTTP URL.

Reply:
Doesn't the URL come from the Server originally anyway ?

Reply:
It comes from the Nexus handler. But because there is no Nexus client, for the tests I have to start the handler workflow manually and specify the callback in that start request.